examples/nextjs/src/graphql/schema/index.ts

```diff
@@ -0,0 +1,308 @@
+import gql from 'graphql-tag';
+import * as Urql from 'urql';
+export type Maybe<T> = T | null;
+export type InputMaybe<T> = Maybe<T>;
+export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
+export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
+export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
+export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
+/** All built-in and custom scalars, mapped to their actual values */
+export type Scalars = {
+  ID: string;
+  String: string;
+  Boolean: boolean;
+  Int: number;
+  Float: number;
+};
+
+export type Mutation = {
```

Comment:
Hey!

I'm out of the loop so I might be missing something, but my suggestion would be to use interfaces for object like types rather than `type`

https://blog.logrocket.com/types-vs-interfaces-in-typescript/

Reply:
This is auto generated by codegen based on your schema and your documents!

Reply:
Got it, weird that they go with `type` then. Do we paste this in or is it a generated file?

Reply:
It's generated, but this behaviour is configurable: https://www.graphql-code-generator.com/plugins/typescript

Reply:
And about the interface or type, I honestly think that is pointless configuration for something that is autogenerated. But I may be missing some positives of the switch.

Reply:
It's my understanding the way this was built was so that types could easily be composed from using `type`. You can change this behaviour, but it's probably not wise to diverge from the defaults for an example like this. After all, the focus of this example should be on showing Grafbase as best we can.

Reply:
What do you think on using a counter example (classic) instead of a todo, so you can focus more on the "infra" than on the pointless components that you need for a todo @notrab

Reply:
It's a tough balance because one would argue that without any minimal styling it doesn't show us off in the best way, and using a common lib like what Supabase did with their UI package for common layout components was more confusing to users.

Either way as long as we maintain these, keep the Grafbase stuff as lean as possible, then it shouldn't be an issue.

With GraphCMS I added minimal styling to these: https://github.com/GraphCMS/graphcms-examples but we got a lot of feedback that they would be better with styling. As long as it's clear what's used and why, I don't think it's an issue to add styling.

I'd then look to build more advanced "reference" type apps similar to https://github.com/GraphCMS/graphcms-commerce-starter and the Vercel projects.